openhtf/plugs/user_input.py

```diff
@@ -96,14 +97,32 @@ def run(self):
         print self._message
 
         # Before reading, clear any lingering buffered terminal input.
-        termios.tcflush(sys.stdin, termios.TCIFLUSH)
+        if sys.stdin.isatty():
+          termios.tcflush(sys.stdin, termios.TCIFLUSH)
 
+        line = ''
         while not self._stopped:
           inputs, _, _ = select.select([sys.stdin], [], [], 0.001)
           for stream in inputs:
             if stream is sys.stdin:
-              response = sys.stdin.readline().rstrip()
-              self._callback(response)
+              new = os.read(sys.stdin.fileno(), 1024)
+              if not new:
+                # Hit EOF!
+                if not sys.stdin.isatty():
+                  # We're running in the background somewhere, so the only way
+                  # to respond to this prompt is the UI. Let's just wait for
+                  # that to happen now. We'll give them a week :)
+                  print "Waiting for a non-console response."
```

Comment:
Where will the print go if we're in the background? Is the (undocumented) assumption that the user will redirect stdout? Maybe consider logging instead of printing?